### PR TITLE
SISRP-17255 - Retrofit CanvasLti::Egrades

### PR DESCRIPTION
**Pull request by redconfetti**
https://jira.berkeley.edu/browse/SISRP-17255

### CODE REVIEW

File: spec/models/canvas_lti/egrades_spec.rb [hunk 1]
```diff
@@ -178,41 +178,42 @@
...
+      expect(result[1][:current_grade]).to eq 'A-'
 
-      # Student with Grade
-      expect(result[5][:sis_login_id]).to eq '4000272'
-      expect(result[5][:final_grade]).to eq 'F-'
-      expect(result[5][:current_grade]).to eq 'F-'
+      expect(result[2][:sis_login_id]).to eq '4000199'
+      expect(result[2][:final_grade]).to eq 'B-'
+      expect(result[2][:current_grade]).to eq 'B-'
     end
```
Comment (paulkerschen) [inline spec/models/canvas_lti/egrades_spec.rb]: @redconfetti I don't know this tool well, but is it still possible to have a case where `final_grade` differs from `current_grade`? If so, would it be worth throwing in one such example?

**Comment by paulkerschen** [comment]
Looks good; left one question about test cases.

File: spec/models/canvas_lti/egrades_spec.rb (hunk 1 above)
Comment (redconfetti) [inline spec/models/canvas_lti/egrades_spec.rb]: @paulkerschen Sure. I'll expand it to have that now.

Comment (redconfetti) [inline spec/models/canvas_lti/egrades_spec.rb]: @paulkerschen There actually is a scenario where the current grade differs from the final grade. One of them (current or final) does not factor in ungraded assignments into the calculation. I forget which one.

**Comment by paulkerschen** [comment]
Thanks @redconfetti!

**Pull request merged**